stdlib/tempfile.pyi

```diff
@@ -547,6 +553,10 @@ class SpooledTemporaryFile(IO[AnyStr], _SpooledTemporaryFileBase):
     def tell(self) -> int: ...
     def truncate(self, size: int | None = None) -> None: ...  # type: ignore[override]
     @overload
+    def _file(self) -> io.BytesIO: ...
+    @overload
+    def _file(self) -> io.TextIOWrapper: ...
```

Comment:
This looks better, although it is missing two things:
- Specifying which of the two overloads a type checker should choose, similar to `write()` below
- The `@property` decorator

That said, I just realized it is not possible to use `@overload` and `@property` together like I imagined. This unfortunately means there's no way to say "this attribute is `BytesIO` for `SpooledTemporaryFile[bytes]` and `TextIOWrapper` for `SpooledTemporaryFile[str]`".

I think the best we can do is using `IO` with a normal attribute, not a property:

```suggestion
    _file: IO[AnyStr]
```

Here `IO[AnyStr]` means that it will be `IO[str]` for `SpooledTemporaryFile[str]`, and `IO[bytes]` for `SpooledTemporaryFile[bytes]`. This will be good enough for most use cases, but not ideal, because `IO` is not as specific as `TextIOWrapper`/`BytesIO`. For example, `some_string_tempfile._file.buffer` will not type check, because even though `TextIOWrapper` has a `buffer` attribute, the more general `IO[str]` does not have it and the type checker complains.

Reply:
Neat. I will accept this PR with a simple `_file: IO[str]` or with the descriptor workaround, since either way it is much better than not having a `_file` attribute at all :)